charts/prefect-worker/templates/deployment.yaml

```diff
@@ -19,9 +19,11 @@ spec:
       app.kubernetes.io/component: worker
   template:
     metadata:
-      {{- if .Values.worker.podAnnotations }}
-      annotations: {{- include "common.tplvalues.render" (dict "value" .Values.worker.podAnnotations "context" $) | nindent 8 }}
-      {{- end }}
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
```

Comment:
the Deployment needs to re-roll in order for the `initContainer` to run and sync the template

this checksum annotation will ensure that the Deployment will re-roll if the template values change (and cause the resulting configmap's hash to change)

Reply:
however, im now realizing that if the end user has a BYO configmap, this checksum wont change.  maybe we just force a re-roll on every `helm upgrade`?

Reply:
Mm, good point. I don't know of a way off the top of my head to handle that condition. Forcing a re-roll on every `helm upgrade` feels a little disruptive but definitely an option.

We could always start with what you currently have, and if someone brings their own ConfigMap, they could use a tool like [reloader](https://github.com/stakater/Reloader).

Reply:
yeah i like that

Reply:
I use reloader and it is a perfect tool to handle pod updates when secrets/configmap changes. No need to handle it within your helm/app.